### ISSUE TITLE
Bug Report: Key error following documentation with V 0.12.3a3

### ISSUE THREAD

**Issue by Slurpgoose** [matplotlib/mplfinance#104]
Hi, I just started using this library to quickly view some json files I had of OHLC Data.
when formatting my data as 
`df = df[['date', 'open', 'high', 'low', 'close']]`

I get a KeyError: 'Open' not in index"

when changing my Dataframe column labels to be uppercase I get a new error thrown
`KeyError: "['date' 'open' 'high' 'low' 'close'] not in index"` 

I solved this issue by monkey patching [_arg_validators.py](https://github.com/matplotlib/mplfinance/blob/469464e2077be5b0758dddb51ee316aa22079fda/src/mplfinance/_arg_validators.py#L31)

through creating this issue, I realized it seems to be fixed in the newest source but does not exist in the pip installed version.

I am on version `mplfinance==0.12.3a3` and this is what im seeing from my traceback
 
        dates   = mdates.date2num(data.index.to_pydatetime())
        opens   = data['Open'].values
        highs   = data['High'].values
        lows    = data['Low'].values
        closes  = data['Close'].values

if any more information is required I will be happy to provide.

thanks

**Comment by DanielGoldfarb** [comment]
@Slurpgoose ,
Hi.  It's not clear to me exactly where the exception is being raised.  Can you please provide your code, and the data, so that I replicate the problem?  Thank you.   --Daniel

**Comment by Slurpgoose** [comment]
@DanielGoldfarb, thanks for the fast response.

while creating the gist to demonstrate the issue I realized the mistake was in my own code. when I changed the column names from lowercase to uppercase I did not update all instances and therefore was getting an error locally.

I have attached a gist since I finished creating it before recognizing my mistake...
https://gist.github.com/Slurpgoose/d5ed03e8aa5399cfcd0b7e6737dcc988

I apologize for taking your time the library works perfectly :)

thanks again
